pkg/cli: TestDemoLocality breaks TestCustomQuery

If `TestDemoLocality` and `TestCustomQuery` run in the same test shard, the former causes the latter to unconditionally fail. This is because `TestDemoLocality` calls `securityassets.ResetLoader()` without resetting the loader back to its previous value.

This can be reproduced with: 

```
bazel test pkg/cli:cli_test  --test_filter 'TestCustomQuery|TestDemoLocality' --test_sharding_strategy disabled
```

See test PR here (https://github.com/cockroachdb/cockroach/pull/127285) that adds a simple TestFoo function to `pkg/cli/flags_test.go` resulting in the following test failure:

```
=== RUN   TestCustomQuery
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCustomQuery1928162877
    test_log_scope.go:81: use -show-logs to present logs inline
    testcluster.go:373: problem using security settings: no certificates found; does certs dir exist?
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCustomQuery1928162877
--- FAIL: TestCustomQuery (0.02s)
```

See: https://mesolite.cluster.engflow.com/invocations/default/5ee36841-d586-4b53-b167-6f1f2f4af5eb


Jira issue: CRDB-40420

--- a/pkg/cli/demo_locality_test.go
+++ b/pkg/cli/demo_locality_test.go
@@ -40,6 +40,7 @@ func TestDemoLocality(t *testing.T) {
 	// asset loader that is set by default in tests will not be able to
 	// find the certs that demo sets up.
 	securityassets.ResetLoader()
+	defer ResetTest()
 
 	// Using datadriven allows TESTFLAGS=-rewrite.
 	datadriven.RunTest(t, datapathutils.TestDataPath(t, "demo", "test_demo_locality"), func(t *testing.T, td *datadriven.TestData) string {

--- a/pkg/cli/demo_test.go
+++ b/pkg/cli/demo_test.go
@@ -34,6 +34,7 @@ func TestDemo(t *testing.T) {
 	// asset loader that is set by default in tests will not be able to
 	// find the certs that demo sets up.
 	securityassets.ResetLoader()
+	defer ResetTest()
 
 	// Using datadriven allows TESTFLAGS=-rewrite.
 	datadriven.RunTest(t, datapathutils.TestDataPath(t, "demo", "test_demo"), func(t *testing.T, td *datadriven.TestData) string {

--- a/pkg/cli/main_test.go
+++ b/pkg/cli/main_test.go
@@ -8,7 +8,7 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-package cli_test
+package cli
 
 import (
 	"os"
@@ -24,6 +24,10 @@ import (
 )
 
 func init() {
+	ResetTest()
+}
+
+func ResetTest() {
 	securityassets.SetLoader(securitytest.EmbeddedAssets)
 }
 

